File: myintent/src/main/java/pl/mareklangiewicz/myintent/RERulesTouchHelperCallback.java
package pl.mareklangiewicz.myintent;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

/**
 * Created by Marek Langiewicz on 22.10.15.
 */
public class RERulesTouchHelperCallback extends ItemTouchHelper.SimpleCallback {

    private final RERulesAdapter mAdapter;

    public RERulesTouchHelperCallback(RERulesAdapter adapter) {
        super(ItemTouchHelper.UP | ItemTouchHelper.DOWN, ItemTouchHelper.START | ItemTouchHelper.END);
        mAdapter = adapter;
    }

    @Override public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
        int pos1 = viewHolder.getAdapterPosition();
        int pos2 = target.getAdapterPosition();
        return mAdapter.move(pos1, pos2);
    }

    @Override public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
        int pos = viewHolder.getAdapterPosition();
        mAdapter.remove(pos);
    }

}

File: myintent/src/main/java/pl/mareklangiewicz/myintent/RERulesTouchHelperCallback.kt
package pl.mareklangiewicz.myintent

import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView


/**
 * Created by Marek Langiewicz on 22.10.15.
 */
class RERulesTouchHelperCallback(private val adapter: RERulesAdapter)
: ItemTouchHelper.SimpleCallback(ItemTouchHelper.UP or ItemTouchHelper.DOWN, ItemTouchHelper.START or ItemTouchHelper.END) {

    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean
            = adapter.move(viewHolder.adapterPosition, target.adapterPosition)

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) = adapter.remove(viewHolder.adapterPosition)
}
